go/cmd/dolt/commands/revert.go

```diff
@@ -79,89 +85,66 @@ func (cmd RevertCmd) Exec(ctx context.Context, commandStr string, args []string,
 		return 1
 	}
 
-	if dEnv.IsLocked() {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(env.ErrActiveServerLock.New(dEnv.LockFile())), help)
-	}
-
-	_, sqlCtx, closer, err := cliCtx.QueryEngine(ctx)
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
-		cli.PrintErrln(err.Error())
+		cli.Println(err.Error())
 		return 1
 	}
-	defer closer()
-
-	headCommit, err := dEnv.HeadCommit(sqlCtx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	headRoot, err := headCommit.GetRootValue(sqlCtx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	workingRoot, err := dEnv.WorkingRoot(sqlCtx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	headHash, err := headRoot.HashOf()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	workingHash, err := workingRoot.HashOf()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	if !headHash.Equal(workingHash) {
-		cli.PrintErrln("You must commit any changes before using revert.")
-		return 1
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
-	headRef := dEnv.RepoState.CWBHeadRef()
-	commits := make([]*doltdb.Commit, apr.NArg())
-	for i, arg := range apr.Args {
-		commitSpec, err := doltdb.NewCommitSpec(arg)
-		if err != nil {
-			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-		}
-		commit, err := dEnv.DoltDB.Resolve(sqlCtx, commitSpec, headRef)
+	var author string
+	if apr.Contains(cli.AuthorParam) {
+		author, _ = apr.GetValue(cli.AuthorParam)
+	} else {
+		name, email, err := env.GetNameAndEmail(cliCtx.Config())
 		if err != nil {
-			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			cli.Println(err.Error())
+			return 1
 		}
-		commits[i] = commit
+		author = name + " <" + email + ">"
```

Comment:
Nit/Q:
Why do manual concatenation here instead of `fmt.Sprintf`?
Also, since the author information is not available for SQL sessions and our commands need to start passing this information more often, should this be a method? `env.GetAuthor` or something?